hera_mc/cm_redis_corr.py

```diff
@@ -0,0 +1,160 @@
+# -*- mode: python; coding: utf-8 -*-
+# Copyright 2018 the HERA Collaboration
+# Licensed under the 2-clause BSD license.
+
+"""
+Methods for handling locating correlator and various system aspects.
+"""
+
+from __future__ import absolute_import, division, print_function
+
+import json
+import redis
+import time
+import six
+from . import cm_sysutils
+from .correlator import DEFAULT_REDIS_ADDRESS
+
+REDIS_CORR_HASH = 'corr:map'
+
+
+def snap_part_to_host_input(part, redis_info=None):
+    """
+    Given a part string, eg. 'e2>SNPC000008' it will return the putative hostname
+    and adc number.  If a redis session is supplied and key available it returns the
+    hostname, otherwise it returns the SNAP name.  If None, returns the SNAP name.
+
+    Parameters
+    ----------
+    part : str
+        port>snap part string as returned by cminfo
+    redis_info : None or dict
+        If dict and the key is available it returns the hostname, otherwise SNAP name
+        If None, returns the SNAP name
+
+    Returns
+    -------
+    hostname : str
+        hostname as parsed from input 'part' and redis
+    adc_num : str
+        port ADC number, as parsed from input 'part'
+    """
+    adc, name = part.split('>')
+    adc_num = int(adc[1:]) // 2  # divide by 2 because ADC is in demux 2
+    if redis_info is None:
+        hostname = name
+    else:
+        _x = redis_info['rsession'].hget(redis_info['rhash'], redis_info['rkey'])
+        if _x is None:
+            hostname = name
+        else:
+            hostname = json.loads(_x)[name]
```

Comment:
this line isn't covered in tests...

Reply:
I added a test to catch this line, which caught a potential key issue.

Reply:
great!